functions/system.bash

```diff
@@ -78,49 +70,51 @@ locale_setting() {
     whiptail --title "Change Locale" --msgbox "For the locale change to take effect, please reboot your system now." 10 60
     return 0
   fi
 
   echo -n "$(timestamp) [openHABian] Setting locale based on openhabian.conf... "
+  # shellcheck source=/etc/openhabian.conf disable=SC1091
   source "$CONFIGFILE"
   if is_ubuntu; then
-    cond_redirect locale-gen $locales
+    # shellcheck disable=2154
+    cond_redirect locale-gen "$locales"
   else
     touch /etc/locale.gen
     for loc in $locales; do sed -i "/$loc/s/^# //g" /etc/locale.gen; done
     cond_redirect locale-gen
   fi
   cond_redirect dpkg-reconfigure --frontend=noninteractive locales
-  cond_redirect LANG=$system_default_locale; export LANG &>/dev/null
-  cond_redirect LC_ALL=$system_default_locale; export LC_ALL &>/dev/null
-  cond_redirect LC_CTYPE=$system_default_locale; export LC_CTYPE &>/dev/null
-  cond_redirect LANGUAGE=$system_default_locale; export LANGUAGE &>/dev/null
-  cond_redirect update-locale LANG=$system_default_locale LC_ALL=$system_default_locale LC_CTYPE=$system_default_locale LANGUAGE=$system_default_locale
-  if [ $? -eq 0 ]; then echo "OK"; else echo "FAILED"; fi
+  cond_redirect "LANG=${system_default_locale:-en_US.UTF-8}"
+  cond_redirect "LC_ALL=${system_default_locale:-en_US.UTF-8}"
+  cond_redirect "LC_CTYPE=${system_default_locale:-en_US.UTF-8}"
+  cond_redirect "LANGUAGE=${system_default_locale:-en_US.UTF-8}"
+  export LANG LC_ALL LC_CTYPE LANGUAGE
+  if cond_redirect update-locale "LANG=$system_default_locale LC_ALL=$system_default_locale LC_CTYPE=$system_default_locale LANGUAGE=$system_default_locale"; then echo "OK"; else echo "FAILED"; fi
 }
 
 hostname_change() {
   echo -n "$(timestamp) [openHABian] Setting hostname of the base system... "
   if [ -n "$INTERACTIVE" ]; then
-    new_hostname=$(whiptail --title "Change Hostname" --inputbox "Please enter the new system hostname (no special characters, no spaces):" 10 60 3>&1 1>&2 2>&3)
-    if [ $? -ne 0 ]; then return 1; fi
+    if ! new_hostname=$(whiptail --title "Change Hostname" --inputbox "Please enter the new system hostname (no special characters, no spaces):" 10 60 3>&1 1>&2 2>&3); then return 1; fi
     if ( echo "$new_hostname" | grep -q ' ' ) || [ -z "$new_hostname" ]; then
       whiptail --title "Change Hostname" --msgbox "The hostname you've entered is not a valid hostname. Please try again." 10 60
       echo "FAILED"
       return 1
     fi
   else
+    # shellcheck source=/etc/openhabian.conf disable=SC1091
     source "$CONFIGFILE"
-    new_hostname="$hostname"
+    new_hostname="${hostname:-openhab}"
   fi
   hostnamectl set-hostname "$new_hostname" &>/dev/null
   hostname "$new_hostname" &>/dev/null
   echo "$new_hostname" > /etc/hostname
-  local TMP="$(mktemp /tmp/.XXXXXXXXXX)"
-  sed "s/127.0.1.1.*/127.0.1.1 $new_hostname/g" /etc/hosts >${TMP} && cp $TMP /etc/hosts
-  rm -f $TMP
+  TMP="$(mktemp /tmp/.XXXXXXXXXX)"
```

Comment:
Should we change this to `/tmp/openhabian.XXXXX` as we discussed earlier?

Reply:
yep, done